### PR TITLE
Ticker api sync

### PR DESCRIPTION
**Pull request by PingvinAustr**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by EyR1oN — APPROVED** [review]

**Review by dimasster — APPROVED** [review]

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 1]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
...
-const tickerItems = [
-    "укладачі Інна Крупник, Оксана Юркова,",
-    "редакторка Ім`я Прізвище,",
-    "ілюстратор Сергій Федоров,","дизайнери Марія Кравцова, Олександр Івашко,",
-    "актор дубляжу Ім`я Прізвище,"
-];
-
 const TickerComponent = () => {
     const streetcodeId = useParams<{id: string}>();
     const id = parseInt(streetcodeId.id ?? "1");
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:10]: Please utilize `useRouteId` hook here.

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 2]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
...
     const {value} = useAsync(() =>subtitlesApi.getSubtitlesByStreetcodeId(id));
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:11]: Please destructure the `getSubtitlesByStreetcodeId` member.

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 3]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
...
     let subtitles = value as Subtitle[];
-    console.log(subtitles);
 
     const createSubtitleString = (subtitles?: Subtitle[]): Map<number, string> => (
         new Map(subtitles?.map(({ id, firstName, lastName, subtitleStatus }) => [id, `${Object.keys(SubtitleStatus)[subtitleStatus]} ${firstName} ${lastName}, `] )) );
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:15]: This function shouldn't be inside of a component.

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 4]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
...
-    "редакторка Ім`я Прізвище,",
-    "ілюстратор Сергій Федоров,","дизайнери Марія Кравцова, Олександр Івашко,",
-    "актор дубляжу Ім`я Прізвище,"
-];
-
 const TickerComponent = () => {
     const streetcodeId = useParams<{id: string}>();
     const id = parseInt(streetcodeId.id ?? "1");
     const {value} = useAsync(() =>subtitlesApi.getSubtitlesByStreetcodeId(id));
     let subtitles = value as Subtitle[];
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:12]: Why would you use `let` declarator here?

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 5]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
 import subtitlesApi from "@api/additional-content/subtitles.api";
 import Subtitle, {SubtitleStatus} from "@models/additional-content/subtitles.model";
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:6]: Fix curly bracket spacing. Everywhere.

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 6]
```diff
@@ -5,34 +5,19 @@ import {useAsync} from "@hooks/stateful/useAsync.hook";
...
 const TickerComponent = () => {
     const streetcodeId = useParams<{id: string}>();
     const id = parseInt(streetcodeId.id ?? "1");
     const {value} = useAsync(() =>subtitlesApi.getSubtitlesByStreetcodeId(id));
     let subtitles = value as Subtitle[];
-    console.log(subtitles);
 
     const createSubtitleString = (subtitles?: Subtitle[]): Map<number, string> => (
         new Map(subtitles?.map(({ id, firstName, lastName, subtitleStatus }) => [id, `${Object.keys(SubtitleStatus)[subtitleStatus]} ${firstName} ${lastName}, `] )) );
     const subtitleMap = createSubtitleString(subtitles);
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:16]: Consider memoizing this expensive operation.
**__REMINDER__**: Do not postpone this as you'll likely won't be working on this component in the nearest future.

**Review by AleXLaeR — CHANGES_REQUESTED** [review]

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 1 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 2 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 3 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 4 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Changed

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 6 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 5 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx [hunk 7]
```diff
@@ -1,19 +1,20 @@
...
+const createSubtitleString = (subtitles?: Subtitle[]): Map<number, string> => (
+    new Map(subtitles?.map(( { id, firstName, lastName, subtitleStatus } ) => [id, `${ Object.keys(SubtitleStatus)[subtitleStatus] } ${ firstName } ${ lastName }, `])));
 
-    const createSubtitleString = (subtitles?: Subtitle[]): Map<number, string> => (
-        new Map(subtitles?.map(({ id, firstName, lastName, subtitleStatus }) => [id, `${Object.keys(SubtitleStatus)[subtitleStatus]} ${firstName} ${lastName}, `] )) );
-    const subtitleMap = createSubtitleString(subtitles);
+const TickerComponent = () => {
+    const id = useRouteId();
+    const { getSubtitlesByStreetcodeId } = subtitlesApi;
+    const { value } = useAsync(() => getSubtitlesByStreetcodeId(id));
```
Comment (AleXLaeR) [inline src/features/StreetcodePage/TickerBlock/Ticker.component.tsx:15]: `useAsync` call should be invoked every time `id` changes.

**Review by AleXLaeR — CHANGES_REQUESTED** [review]

File: src/features/StreetcodePage/TickerBlock/Ticker.component.tsx (hunk 7 above)
Comment (PingvinAustr) [reply src/features/StreetcodePage/TickerBlock/Ticker.component.tsx]: Done

**Pull request merged**